frappe/integrations/doctype/ldap_settings/ldap_settings.py

```diff
@@ -142,6 +148,61 @@ def get_ldap_attributes(self):
 
 		return ldap_attributes
 
+
+	def fetch_ldap_groups(self, user, conn):
+		fetch_ldap_groups = None
+
+		ldap_attributes = ['cn']
+
+		ldap_object_class = None
+		ldap_group_members_attribute = None
+		ldap_group_search_filter = None
+
+
+		if self.ldap_directory_server.lower() == 'active directory':
+
+			ldap_object_class = 'Group'
+			ldap_group_members_attribute = 'member'
+			user_search_str = user.entry_dn
+
+
+		elif self.ldap_directory_server.lower() == 'openldap':
+
+			ldap_object_class = 'posixgroup'
+			ldap_group_members_attribute = 'memberuid'
+			user_search_str = getattr(user, self.ldap_username_field).value
+
+		elif self.ldap_directory_server.lower() == 'custom':
+
+			ldap_object_class = self.ldap_group_objectclass
+			ldap_group_members_attribute = self.ldap_group_member_attribute
+			user_search_str = frappe.as_unicode(frappe.form_dict.usr)
```

Comment:
fix:

a user object is passed to this function `user` don't reach outside of function to get user details.

Reply:
implemented in c0565ae2d3d8858d728d391e8a2163e6c071a601